### PR TITLE
Define `SignatureVerifier` API

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Adds SPI types for the Hedera signature verification facility.
 - ⚠️  Possibly controversial: Adds SPI dependency on `com.swirlds.common` to re-use platform's `VerificationStatus` enum.
 - ⚠️ Also provocative: Claims in javadoc that there are three instances where business logic will at least participate in enforcing signing requirements (within an EVM system contract, before triggering a scheduled transaction, and evaluating the top-level signature on a `FileDelete`).

### CODE REVIEW

**Review by mhess-swl — APPROVED** [review]
A question did come up about depending on the swirlds common library, which may still be an issue, but looks good!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by povolev15 — APPROVED** [review]
LGTM

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/signatures/SignatureVerifier.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Service} business logic must detect and enforce the second and third checkpoints.
+ */
+public interface SignatureVerifier {
+    /**
+     * Returns whether the given key has signed at the current checkpoint.
+     *
+     * @param key the key to test
+     * @return whether the key has signed
+     */
+    boolean hasSigned(@NonNull HederaKey key);
```
Comment (netopyr) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/signatures/SignatureVerifier.java:44]: I am afraid, I do not really understand how this works. Is there some documentation of the overall design?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/signatures/SignatureVerifier.java:44]: I will add a flow diagram when I am implementing it this week @netopyr

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**